src/travel-details-map-screen/TravelDetailsMapScreenComponent.tsx

```diff
@@ -147,42 +158,66 @@ const styles = StyleSheet.create({
 
 type VehicleIconProps = {
   coordinates: Coordinates;
+  mode?: TransportMode;
+  subMode?: TransportSubmode;
   setShouldTrack: React.Dispatch<React.SetStateAction<boolean>>;
 };
 
-const LiveVehicle = ({coordinates, setShouldTrack}: VehicleIconProps) => {
+const LiveVehicle = ({
+  coordinates,
+  setShouldTrack,
+  mode,
+  subMode,
+}: VehicleIconProps) => {
+  const circleColor = useTransportationColor(mode, subMode);
+  const iconName = vehicleIconName(mode);
+
   if (!coordinates) return null;
   return (
     <MapboxGL.ShapeSource
-      id="vehicle"
+      id="liveVehicle"
       shape={pointOf(coordinates)}
       cluster
       onPress={() => {
         setShouldTrack(true);
       }}
     >
-      <MapboxGL.SymbolLayer
-        id="icon"
-        minZoomLevel={1}
+      <MapboxGL.CircleLayer
+        id="liveVehicleCircle"
+        minZoomLevel={8}
         style={{
-          iconImage: {uri: 'ClusterCount'},
-          iconSize: 2,
-          iconAllowOverlap: true,
-          iconAnchor: 'center',
+          circleColor,
+          circleRadius: 22,
         }}
       />
       <MapboxGL.SymbolLayer
-        id="clusterCount"
-        minZoomLevel={1}
-        aboveLayerID="icon"
+        id="liveVehicleIcon"
+        aboveLayerID="liveVehicleCircle"
+        minZoomLevel={8}
         style={{
-          textField: 'BUS',
-          textColor: 'black',
-          textSize: 11,
-          textAnchor: 'center',
-          textAllowOverlap: true,
+          iconImage: {uri: iconName},
```

Comment:
Could you elaborate a little about using images uploaded to MapBox instead of the transportation icons from the design system lib?

Reply:
You can't inject svg data from the app (as far as I'm aware), so these images are manually uploaded in Mapbox Studio. `iconName` is a reference to this upload, not the design system. I can add a comment explaining this 👍 

I've only uploaded a subset of transport icons, since only a few is likely to appear with position data.

Reply:
Aight. I looked at `SymbolLayer` doc now, and it seems pretty cumbersome to do it there. When reading the RN Mapbox docs it seemed like maybe [MarkerView](https://github.com/rnmapbox/maps/blob/main/docs/MarkerView.md) would be a more appropriate solution, but not necessary to go there now if we feel this works good enough.

Reply:
I think this works for now, but that might change as soon as we look at the direction arrow. Will look at MarkerView when I get to that 👍

Reply:
Sounds as a good plan! Just by reading about MarkerView it seems like a good choice for this kind of feature.